src/AlertBar/Icon.js

```diff
@@ -0,0 +1,49 @@
+import React, { isValidElement } from 'react'
+import propTypes from 'prop-types'
+import { Valid, Warning, Error, Info } from '../icons/Status'
+import { spacers } from '../theme.js'
+
+const Icon = ({ icon, success, warning, critical }) => {
+    if (icon === false) {
+        return null
+    }
+
+    let IconComponent
```

Comment:
The naming doesn't really match the convention. The assigned value is not a component anymore but already the computed element (component would be `Error` while the element is `<Error />`), which is why I think this one needs to be camelCase (plus the name `Component` is confusing, I'd expect that the jsx code will do `<IconComponent />`.

React only enforces the first letter to be capitalized when using the `<... />` style, not when doing `{component}`.